central/notifiers/syslog/tcp_sender.go

```diff
@@ -82,7 +83,17 @@ func validateRemoteConfig(endpointConfig *storage.Syslog_TCPConfig) (string, err
 		return "", errors.Errorf("invalid port number %d must be between 1 and 65535", port)
 	}
 
-	return fmt.Sprintf("%s:%d", endpointConfig.GetHostname(), endpointConfig.GetPort()), nil
+	hostName := fmt.Sprintf("%s:%d", endpointConfig.GetHostname(), endpointConfig.GetPort())
+
+	sysURL := fmt.Sprintf("tcp://%s", hostName)
+
+	_, err := url.ParseRequestURI(sysURL)
+
+	if err != nil {
+		return "", errors.New("invalid host name")
```

Comment:
We get a returned `err` value from the parse function, we might want to pass that up along with our new one:
```
return "", errors.Wrap(err, "invalid request url")
```